cypress/support/Helpers.js

```diff
@@ -21,31 +21,22 @@ export function RemoveDuplicates(inputArray) {
   return uniqueOutputArray
 }
 
-export function RemoveMarkup(stringWithHtml) {
-  var tempDocument = document.createElement("div")
-  tempDocument.innerHTML = stringWithHtml
-  var stringToReturn = tempDocument.textContent || tempDocument.innerText || ''
-  stringToReturn.replace('\u200B', '') // zero width space
-  stringToReturn = stringToReturn.trim()
-  return stringToReturn
-}
-
-export function StringArrayFromInnerHtml(selector, removeMarkup = true) {
+export function StringArrayFromElementText(selector, retainMarkup = false) {
   var elements = Cypress.$(selector)
-  ConLog(`StringArrayFromInnerHtml(${selector})`, elements.length)
+  ConLog(`StringArrayFromElementText(${selector})`, elements.length)
   var returnValues = new Array()
   for (var i = 0; i < elements.length; i++) {
-    if (removeMarkup) returnValues.push(RemoveMarkup(elements[i].innerHTML))
-    else returnValues.push(elements[i].innerHTML)
-    ConLog(`StringArrayFromInnerHtml(${selector})`, returnValues[i])
+    if (retainMarkup) returnValues.push(elements[i].innerHTML)
+    else returnValues.push(elements[i].innerText)
+    ConLog(`StringArrayFromElementText(${selector})`, returnValues[i])
   }
   return returnValues
 }
```

Comment:
Thanks Matt. I agree that this is better, I did not know about the map method... I'll make that change tomorrow.

FYI - You are really pushing me to do better with JavaScript. Usually when I am learning something new on the job I am working in an existing code base where I have existing code to draw on and that teaches me, but in this case I've created most of this code myself and I only search for functions and patterns when I don't know how to do them. I've had to put in a couple of extra hours on days where you give me these suggestions, I'm not billing MS for that time however since I am gaining the experience.

Reply:
Did I understand you correctly, do you prefer the 1st to the 2nd...
`var anArray = [];`
`var anArray = new Array();`
If so why?

Reply:
We should probably have a voice discussion about camelCasing the function names. I'm open minded to it but, I have questions and one issue.

Also if you know of a good place that details the standards you want me to follow I'd be willing to give it a read on my on time.

Reply:
Yes, although it would be `let` not `var`.  I saw in a previous location you used an object literal `{}` instead of `new Object()`. This is the same idea to use an array literal.

The main reason is just convention and readability. I think there are some compilers that optimize better for literals also but we're not worried about that.  

An analogy would be like if you went to France and kept speaking English. It's still a latin based language like C# and Javascript are to C but it's still different.  You can argue that you disagree with the creators or French or whatever, but that's how it is and that's how people expect to communicate. I view this similarly as trying to write in a way other javascript developers will expect. There are personal stylistic differences but for the most part these are accepted and have a reason for them.

Also, when you say 
> You are really pushing me to do better with JavaScript.

These patterns aren't specific to javascript. Trying to be as simple, clear, and expressive as possible to what the code is actually doing should be uniform across all languages. Eg. not using global variables, reducing imperative code, making code self-document and be robust to unintended use, etc.  Other than casing you would follow similar patterns in C#, Java, Python,

Reply:
Points taken and agreed with except...

I came into this project not knowing JavaScript and not knowing Cypress. JavaScript was easy enough, but I still had to look up a lot of things as I worked. Cypress on the other hand was quite difficult as you and my predecessor discovered when you both worked on it. If I could code in Cypress without any logic, just pure Cypress then it would be easy, but I really don't know how to avoid the logic, in the beginning I really did try.

**In my first 6 weeks I put in a LOT of extra time that I did not bill for to come up to speed and accomplish what I have in a short time.** 

So my focus was to get as much test coverage as possible in the shortest amount of time. Both for the sake of the project and for the sake of keeping my job and not meeting the same fate as my predecessor.

While I have not been using all of the preferred JavaScript ways, I have written good solid code that is easy to read. I believe it is well structured. Now that I am settled in I am willing to put in even more of my personal time to come up to the JavaScript standards you want to hold me to.

Reply:
Wow... 
This is not the response I expected. I need to work on how I phrase these comments.